api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -996,6 +996,16 @@ type AzureNodePoolPlatform struct {
 	// +optional
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
 
+	// enableEncryptionAtHost enables encryption at host on virtual machines. According to Microsoft documentation, this
+	// means data stored on the VM host is encrypted at rest and flows encrypted to the Storage service. See
+	// https://learn.microsoft.com/en-us/azure/virtual-machines/disks-enable-host-based-encryption-portal?tabs=azure-powershell
+	// for more information.
+	//
+	// +kubebuilder:default:=Enabled
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +optional
+	EnableEncryptionAtHost string `json:"enableEncryptionAtHost,omitempty"`
```

Comment:
nit: `EncryptionAtHost` instead of `EnableEncryptionAtHost` otherwise you'll have "Disabled" "enable encryption at host" which can get confusing.

Reply:
Makes sense. Updated.